### PR TITLE
Port to Win32 (via MSYS2)

### PR DESCRIPTION
**Pull request by steveschnepp**

### CODE REVIEW

File: isp55e0.c [hunk 1]
```diff
@@ -27,7 +27,23 @@
...
+
+static uint32_t be32toh(uint32_t const net) {
+    uint8_t data[4] = {};
+    memcpy(&data, &net, sizeof(data));
+
+    return ((uint32_t) data[3] << 0)
+         | ((uint32_t) data[2] << 8)
+         | ((uint32_t) data[1] << 16)
+         | ((uint32_t) data[0] << 24);
+}
```
Comment (frank-zago) [inline isp55e0.c]: No swap instruction on windows? What about htonl?

File: isp55e0.c [hunk 2]
```diff
@@ -840,10 +840,15 @@ static void load_file(struct device *dev, struct content *info)
...
 	memset(info->buf, 0xff, info->len);
 
-	/* TODO: loop until all read, or use mmap instead. */
-	ret = read(fd, info->buf, statbuf.st_size);
-	if (ret != statbuf.st_size)
-		err(EXIT_FAILURE, "Can't read firmware file");
+	off_t total_read = 0;
+	while (total_read < statbuf.st_size) {
+		int ret = read(fd, info->buf + total_read, statbuf.st_size - total_read);
+		if (ret == 0) break;
```
Comment (frank-zago) [inline isp55e0.c]: If the code hits this, something is wrong. That loops reads exactly the size of the file. It'd treat that as an error too.

File: isp55e0.c (hunk 1 above)
Comment (steveschnepp) [reply isp55e0.c]: Indeed, `_byteswap_ulong` works nicely.

File: isp55e0.c (hunk 2 above)
Comment (steveschnepp) [reply isp55e0.c]: Indeed, but in win32 it ends with a read == 0.
i have to check why, as this should mean that the file is not read in full...

Yet the uploaded code does work on the MCU.

I'll dig that mystery.

Comment (steveschnepp) [reply isp55e0.c]: Ok. never mind. I was stupid. It is solved.

**Comment by steveschnepp** [comment]
@frank-zago should be better now 😉

**Comment by steveschnepp** [comment]
@frank-zago I found out the mysterious bug i had. I was missing `O_BINARY` which is fixed in 7333ce3496b0f483e8a59f2788488176ccdece86.

**Comment by frank-zago** [comment]
That looks good. Can you make just 1 commit for all this?

**Comment by steveschnepp** [comment]
@frank-zago wish granted. You can even merge it now without a merge commit :wink:

**Comment by frank-zago** [comment]
Thank you.

**Pull request merged**